src/aggregate/aggregate_exec.c

```diff
@@ -447,9 +447,7 @@ static void sendChunk_Resp2(AREQ *req, RedisModule_Reply *reply, size_t limit,
     }
 
     // Set `resultsLen` to be the expected number of results in the response.
-    if (ShouldReplyWithTimeoutError(rc, req)) {
-      resultsLen = 1;
-    } else if (rc == RS_RESULT_ERROR) {
+    if (rc == RS_RESULT_ERROR) {
```

Comment:
is this correct also when no documents are returned?

Reply:
can you add a test for it? (with FT.DEBUG FT.DEARCH TIMEOUT_AFTER_N 0) ?

Reply:
The condition is just redundant as far as I can see..
I don't want to change the behavior of possible paths, since that will be breaking

Reply:
Yeah i see the repetition...
So in general,  do we have test with timeout error and 0 results?

Reply:
Not sure
It's hard to say since this requires a traversal of a lot of tests. We can create a ticket for the addition of such a test